Converted Addon to the V2 Format

cah-briangantzler opened the pull request:
Converted addon to a V2 format https://github.com/sinankeskin/ember-inputmask/issues/348
Updated the ember-modifier to use the new modify hook.


I updated some of the github ci paths because of the V2 format, but not sure how to test. Please look over there and see if there is anything else that would need fixed. Im not sure how you release to NPM, but that may also be slightly different due to the V2 format. Please read https://github.com/embroider-build/embroider/blob/main/PORTING-ADDONS-TO-V2.md These are the directions I followed. 

The last line does say
```
To publish, you will run npm publish in the addon directory.
```

sinankeskin commented:
Sorry. I couldn't find free time to take a look. I definitely will on weekend.

sinankeskin commented on addon/src/modifiers/inputmask.js:
```diff
@@ -1,39 +1,49 @@
+    const args = this.getArgs(positional, named);
 
-    this._setInputMask(args);
+    new InputMask(args).mask(element);
```
Looks like you removed _setInputMask invoke here but method still exists. I believe we should revert this line back.

cah-briangantzler replied:
I meant to delete the method. Its a one line method that is invoke in only one place now. Also no longer needs to return.

Its your call. I think deleting the method is the way to go. You want me to add back the invoke?

sinankeskin replied:
Nope I meant if you want to keep it we should revert this line back. If you want to remove totally fine.

sinankeskin commented on addon/src/modifiers/inputmask.js:
```diff
@@ -54,7 +64,7 @@ export default class InputmaskModifier extends Modifier {
 
-  _setInputMask(args) {
-    new Inputmask(args).mask(this.element);
+  _setInputMask(element, args) {
```
We should revert this back with just args parameter IMO.

cah-briangantzler replied:
see previous comment

sinankeskin commented on addon/src/modifiers/inputmask.js:
```diff
@@ -1,39 +1,49 @@
 /* globals Inputmask */
 import Modifier from 'ember-modifier';
 import { getOwner } from '@ember/application';
+import InputMask from "inputmask";
```
Could you please run `yarn:lint:fix` inside the projects?

cah-briangantzler replied:
In all my personal projects I use lint-staged to always run lint and prettier when I commit, so it just happens. I always forget this when doing external projects.

sinankeskin commented on test-app/tests/integration/components/inputmask-test.js:
```diff
@@ -0,0 +1,86 @@
+      this.value = value;
+    };
+
+    debugger;
```
You forgot 3 `debugger` invokes in this file.

cah-briangantzler replied:
thats embarrassing :)

sinankeskin reviewed (changes requested):
Quite big change but overall looks good and works good. Just minor things. Thank you @cah-briangantzler Appreciated.

sinankeskin commented:
We need to change `ci.yml` according to latest changes @cah-briangantzler Tests are failing. Actually doesn't event start.

cah-briangantzler commented:
Yea, I didnt have any real way of simulating the ci locally. Tests were suppose to be able to run by adding the working dir. I thinking because of the yarn workspaces there needs to be a yarn somewhere else. Not that I have a log will try a few things, but may take a couple of attempts

sinankeskin commented:
I don't want give you too much trouble. We could disable for now if you'd like. I can merge and publish because this state is far from I can image. All looks and works great.

cah-briangantzler commented:
Let me have a couple of attempts. You are using `npm ci` which requires a lock file, there isnt a lock file at the root because I used `yarn`. Do you have a preference? I usually use yarm but can switch it over to npm easy enough

sinankeskin commented:
No I don't have any preference. I used `npm` just because it was default. Actually I use `yarn` all the time with my job projects.

sinankeskin commented:
Error: ENOENT: no such file or directory, lstat '/home/runner/work/ember-inputmask/ember-inputmask/addon/./dist/_app_/components/inputmask.js'

Either we need to run `yarn prepublishOnly` first or we need to send `dist` folders too I guess.

cah-briangantzler commented:
We need to insert a 
```
yarn build
```
somewhere Im thinking

cah-briangantzler commented:
Updated and added the `prepare` script. I think that might be it.  Using https://github.com/kaliber5/ember-stargate/blob/master/.github/workflows/ci.yml as an example. I added a working directory and run the tests directly. He has a `test` script at the root that redirects to the `test` in the package. He (or other examples I looked at) do not check in dist. I think the missing piece was https://github.com/kaliber5/ember-stargate/blob/master/packages/ember-stargate/package.json#L29

You will also note this, https://github.com/bgantzler/ember-inputmask/blob/addon-v2-format/addon/package.json#L19 which I believe it what creates the dist for when it is published to npm. That was gotten from here https://github.com/embroider-build/embroider/blob/main/PORTING-ADDONS-TO-V2.md#part-4-convert-addon-to-v2 step 15

sinankeskin commented:
Hey @cah-briangantzler added you as a collaborator to this repo. You can run the test whenever you want now.

cah-briangantzler commented:
WooHoo.

Hope you are able to make the needed changes to release it. Ember-stargate uses release-it and here is some of what he has in the package.json https://github.com/kaliber5/ember-stargate/blob/master/package.json#L31. Maybe that will help as an example

sinankeskin commented:
> 

Awesome job. I don't wanna use another package. I'll try to release with just npm. I'll try to figure it out and let you know.

Thank you @cah-briangantzler Appreciated.

cah-briangantzler commented:
Re-read the first message, it has a link to the docs I used and steps to release

sinankeskin commented:
Got it. Thank you. Merging.  🎉

cah-briangantzler commented:
https://github.com/kaliber5/ember-stargate/blob/master/package.json#L51 seems we may need the version number at the root package.

Attempting to link to packge using #main is not working.

sinankeskin commented:
Makes sense but I just released. 🎉  Next time I'll update. Thank you again @cah-briangantzler
